File: tanaman_herbal_service/app/Models/Plant.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Plant extends Model
{
    use SoftDeletes;

    public $table = 'plants';

    protected $fillable = [
        'id',
        'name',
        'latin_name',
        'advantage',
        'ecology',
        'endemic_information',
        'qrcode',
        'status',
        'habitus_id',
        'created_by',
        'updated_by',
        'deleted_at',
    ];

    protected $dates = ['deleted_at'];

    public function createdBy()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function updatedBy()
    {
        return $this->belongsTo(User::class, 'updated_by');
    }
    public function habituses()
    {
        return $this->belongsTo(Habitus::class, 'habitus_id');
    }

    public function images()
    {
        return $this->belongsToMany(Image::class, 'plant_images');
    }

    public function lands()
    {
        return $this->belongsToMany(Land::class, 'plant_lands');
    }

    public function languages()
    {
        return $this->belongsToMany(Language::class, 'plant_languages')
            ->withPivot('name', 'advantage', 'ecology', 'endemic_information')
            ->withTimestamps();
    }

    public function translate($langCode)
    {
        $language = Language::where('code', $langCode)->first();
        if (! $language) {
            return [
                'name'                => $this->name,
                'advantage'           => $this->advantage,
                'ecology'             => $this->ecology,
                'endemic_information' => $this->endemic_information,
            ];
        }

        $translation = $this->languages()->where('language_id', $language->id)->first();
        return [
            'name'                => $translation?->pivot->name ?? $this->name,
            'advantage'           => $translation?->pivot->advantage ?? $this->advantage,
            'ecology'             => $translation?->pivot->ecology ?? $this->ecology,
            'endemic_information' => $translation?->pivot->endemic_information ?? $this->endemic_information,
        ];
    }
}

File: tsth_app/lib/features/validation/domain/entities/plant.dart
import 'package:tsth_app/features/validation/domain/entities/habitus.dart';

class Plant {
  final int id;
  final String name;
  final String latinName;
  final String advantage;
  final String ecology;
  final String endemicInformation;
  final String qrcode;
  final int status;
  final Habitus? habitus;
  final String? createdAt;
  final String? updatedAt;
  final String? deletedAt;

  Plant({
    required this.id,
    required this.name,
    required this.latinName,
    required this.advantage,
    required this.ecology,
    required this.endemicInformation,
    required this.qrcode,
    required this.status,
    required this.habitus,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
  });
}
